Wasm-container/README.md

```diff
@@ -0,0 +1,509 @@
+| WASM_MOUNT | An optional space-separated list of host directories to mount in the UDx container --- that is, "I'm going to want to refer to files in these directories in, for example, my makefile, or as include directories in code to be compiled" |
+| VWASM_ENV | names a file filled with variable definitions.  This uses the --env-file option of the "docker run" command, so the file needs to be formatted for that option |
+
+If you have multiple vwasm containers, you can
+choose among them by specifying the tag, which
+is derived from `vwasmsdk:${OSTAG}-${VERTICA_VERSION}`, if those
+variables are set in the environment (you can also define `CONTAINER_TAG` if you've chosen a different container tagging scheme).
+
+`vwasm-bash` creates a shell with the following directories mounted, sharing the name they have in the host system:
+
+- `$HOME`
+- your currently connected directory
+- any directories listed in the environment variable `$VWASM_VOLUMES`
+
+In addition, a Docker volume is mounted as `/vwasmdata`.
+
+
+### An annoying thing about Rust
+
+The primary role of the container is to provide a place to install
+those components that require privileges to install.
+
+Rust and wasmer are designed to be installed privately, in one's
+home directory.
+
+While we play games in the Dockerfile with installing those tools in
+`/usr/WebAssembly/template`, and provide a script to copy them into
+one's home directory, the changes don't always seem to stick between
+container invocations.  This manifests for me as the following:
+
+   error: toolchain 'stable-x86_64-unknown-linux-gnu' is not installed
+
+when I try to use `rustc` to compile a rust file to wasm.
+
+So, there is a `./reinstall-rust` command provided to run inside the
+container to run the rust installation commands  
+
+```shell
+./reinstall-rust $WASMHOME
+```
+
+# An experiment with compiling to Wasm
+
+In the container's `/usr/WebAssembly/vertica` directory are some simple files used as a proof-of-concept for creating unfenced Vertica UDxes in C and Rust using Wasm (these files are also in the `examples/UDx` subdirectory of this repository).  
+
+Copy the directory to your own sandbox (so the files are in a directory you can modify).
+
+There is a `Makefile` with commands to build the example programs.
+
+Make targets are:
+
+- `clean`: remove the constructed files
+- `wasmer-hello`: compile a simple "hello, world" C program to Wasm
+- `run_hello`: execute the `wasmer-hello` program by specifying the dynamically-linked library path.
+- `sum.rs.wasm`: compile the Rust module `sum.rs` to a Wasm module that can be linked into a Vertica UDx.
+- `sum.c.wasm`: compile the C `sum.c` module to a Wasm module that can be linked into a Vertica UDx.
+- `udx_wasm.o`: compile the C++ wrapper for Wasm UDxes.
+- `libudx_wasm.a`, `libudx_wasm.so`: static and dynamic libraries containing `udx_wasm.o`
```

Comment:
Up to here these were make commands, but these last two sound like descriptions of files.  ?